### PR TITLE
all: add support for running tests locally

### PR DESCRIPTION
**Pull request by knusbaum**
Adding a docker-compose.yaml and script that runs all the tests in the repository

### CODE REVIEW

**Review by nsrip-dd — DISMISSED** [review]
Thanks Kyle, this is super useful! I have one comment about the helper script, more of an observation than anything.

File: run-all-tests.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+## CORE
+#echo testing core
+#PACKAGE_NAMES=$(go list ./... | grep -v /contrib/)
+#gotestsum --junitfile ./gotestsum-report.xml -- $PACKAGE_NAMES -v  -coverprofile=coverage.txt -covermode=atomic
+
+## wait extra long for all the docker containers to be "ready"
+#echo Waiting for docker for 120 seconds
+#sleep 120
+
+sleep 20
```
[the remark this answers is not in the document]
Comment (knusbaum) [reply run-all-tests.sh]: Yeah, I am not able to get that test to pass no matter how long I wait. I think there is some problem with the setup, but I'm struggling to figure out what.

The other kafka integration passes no problem. This seems to be the final hold-out for getting integration tests running locally.

**Review by nsrip-dd — APPROVED** [review]
Good find fixing the Kafka issue. I tried the new script out a few times and it works for me. Thanks again for putting this together!

**Review by ajgajg1134 — APPROVED** [review]
I'm not very good at bash scripting so it's tough for me to comment on that file specifically but everything else looks good!

**Pull request merged**